Implement Ifriit

Adrian16199 opened the pull request:
The mystagogue's guardian.


So as it stands, this can be merged upon innate tool fix gets merged into master and notin pyrokinesis until it gets reworked.
Converting it to draft and going to apply changes for it to use innate tool again.

<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What did you change in this PR? -->
YML
## Why / Balance
<!-- Why was it changed? Link any discussions or issues here. Please discuss how this would affect game balance. -->
They used to be a thing, simply fixes them
## Media
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
![image](https://github.com/DeltaV-Station/Delta-v/assets/144424013/83b7de86-2c76-40fe-b5d3-1738c5536d33)

https://github.com/DeltaV-Station/Delta-v/assets/144424013/5d1e7ab2-6804-4f03-b593-1d2bc838221e


## Breaking changes
<!--
List any breaking changes, including namespace, public class/method/field changes, prototype renames; and provide instructions for fixing them. This will be pasted in #codebase-changes.
-->

**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->

:cl:
- fix: Implements ifriit

Adrian16199 committed:
Fixes ifriit.

The mystagogue's guardian.

TO-DO: Figure out how to give the ifriit the actual tool instead of bite that still has the damage propertys of the tool?

Adrian16199 committed:
Hands, innatetool DEATH

giving different parent to make the bite attack frick off.
Dont let them shove, simple different attack would make sense.

Adrian16199 committed:
Merge branch 'master' into fix_ifrit

Adrian16199 committed:
forgot to delete that one

Adrian16199 committed:
Merge branch 'fix_ifrit' of https://github.com/Adrian16199/Delta-v-rebase into fix_ifrit

VMSolidus commented:
To make the ifrit spawn with a tool, you need to give them a relevant Inventory Template, so that they have handslots(Preferably ones immune to strip), and then you need a Loadout. The relevant components are

- [ ] LoadoutComponent (Define what items the Ifrit spawns with and in what slots)
- [ ] HandsComponent (They need hands to use a weapon, should be self explanatory)
- [ ] InventoryComponent (This stores a templateId, which lets you define what inventory slots the Ifrit has)
- [ ] InventorySlotsComponent (Lets the player actually see the equipment UI)

To make it so that nobody can take the tool, you just need to make sure that they don't have the StrippableComponent.

Adrian16199 commented:
I though about tryin to restore InnateItemSystem (which i was lookin for a bit) but I suppose trying that could be the right solution in the first place.

VMSolidus commented:
> I though about tryin to restore InnateItemSystem (which i was lookin for a bit) but I suppose trying that could be the right solution in the first place.

Yea, not much reason to make/remake an old component if the options to do what you want to do already exist. Wizden's already done the work for you, you just need to know what components and arguments are necessary.

Adrian16199 commented:
> > I though about tryin to restore InnateItemSystem (which i was lookin for a bit) but I suppose trying that could be the right solution in the first place.
> 
> Yea, not much reason to make/remake an old component if the options to do what you want to do already exist. Wizden's already done the work for you, you just need to know what components and arguments are necessary.

You know, I did saw the starting gear but i never saw inhand option but you reminding me, made me realize that space ninja spawns with inhand item.

Adrian16199 committed:
Just going to put this together I suppose?

VMSolidus commented:
![image](https://github.com/DeltaV-Station/Delta-v/assets/16548818/f87fb9db-ed47-4520-82bd-978704be4eae)
You need to remove the negative damage component here. It will cause the ifrit to directly heal people from their blunt damage if they have any.

Adrian16199 commented:
okay so first things first, the loadout system doesnt spawn stuff in someone's hand and even if you write that in, it spawns *On* you but not in your hand. Space Ninja doesnt even spawn with it in their hand but it just spawns behind them which is just stupid.

Adrian16199 committed:
I guess his hands atleast work

Adrian16199 committed:
whoopsie yaml linter

Guess-My-Name commented:
> ![image](https://private-user-images.githubusercontent.com/16548818/297253115-f87fb9db-ed47-4520-82bd-978704be4eae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU0OTg0NTYsIm5iZiI6MTcwNTQ5ODE1NiwicGF0aCI6Ii8xNjU0ODgxOC8yOTcyNTMxMTUtZjg3ZmI5ZGItZWQ0Ny00NTIwLTgyYmQtOTc4NzA0YmU0ZWFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE3VDEzMjkxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgwYmIyNGY4ZDQ5M2M1NDVlZDI1ODQ0OTFiYzFjZTg0ODYyNDQ3MWVmZmZiOGJiMDQ1NGU4ODNmMWM4NDE2ZjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fDmYTtYfEEhh6BapHwnZxC-dDQqVssrG94RSiJzoWLk) You need to remove the negative damage component here. It will cause the ifrit to directly heal people from their blunt damage if they have any.

It doesn't heal people. Normal welder does 5 blunt damage when turned off, and also has litMeleeDamageBonus component, but welder has 8 extra heat damage and -5 blunt, instead of extra 10 heat like ifrit does. If you remove it, when Ifrit hand is lit, it will deal 10 heat and 5 blunt.

Adrian16199 committed:
Merge branch 'master' into fix_ifrit

Adrian16199 committed:
Finishing touches and cleaning up.

Not sure how long I will have to living but eh, at the least might aswell finish some PR's

Adrian16199 committed:
Merge branch 'master' into fix_ifrit

Adrian16199 committed:
Eh, i wanted to keep it but i guess not.

Adrian16199 committed:
Update misc_startinggear.yml

Adrian16199 committed:
Merge branch 'master' into fix_ifrit

Adrian16199 committed:
Update ifrit.yml

Adrian16199 commented:
Made a video showcasin that it works, made a final adjustment, merged up. 
Pretty much ready.

VMSolidus commented on Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/familiars.yml:
![image](https://github.com/DeltaV-Station/Delta-v/assets/16548818/370bb8dd-ff0e-40a4-a732-319ece5a67f4)

Hey, as a heads up. While I have finished testing Ifrit and have verified that they now work correctly. I have an issue with how the Pyrokinesis power interacts with recent upstream changes to the "On Fire" state. Namely, actually setting people on fire has had its damage buffed into the stratosphere, which has inadvertently made Pyrokinesis into a point and click near-instant death. This screenshot was what I found in testing Ifrit Pyrokinesis, and unfortunately in light of this I cannot recommend merging this PR unless their Pyrokinesis power is commented out. Pyrokinesis will be a subject of an upcoming rework and refactor of psychic powers, in which I will uncomment the line in question when I feel a reworked Pyrokinesis is safe to let the ifrit have. 

![image](https://github.com/DeltaV-Station/Delta-v/assets/16548818/6bb170b9-48d0-41a9-90da-cea1adacc82b)

Adrian16199 replied:
Hm, well..... Im uncertain if there is a need to rush merging ifriit back in since they arent really requested by anyone from what im aware and mystagogues dont seem to complain or atleast not do it loudly.
I would personally be fine just keeping this waitin until the rework of the powers is here, if you would be fine with said option of course.

DebugOk reviewed (dismissed):
This approval means nothing besides removing my review requirement. When @VMSolidus considers it good enough it can be merged

Adrian16199 committed:
Merge branch 'master' into fix_ifrit

Adrian16199 committed:
Merge branch 'master' into fix_ifrit

DebugOk commented:
Inate Tools got fixed, so that can be used again

Adrian16199 committed:
Merge branch 'master' into fix_ifrit

Adrian16199 commented:
so innate tool does indeed work but weirdly enough, only one hand appears, probably an issue on my end so will try to get it done and just push the current changes to make it less messy.

Adrian16199 committed:
Innate tooling...

Adrian16199 committed:
Makes it work just how its needed to do so.

Adrian16199 committed:
brings back the previous pngs

Adrian16199 committed:
Merge branch 'master' into fix_ifrit

Adrian16199 commented:
Works ingame, changed everything neccesary and removed everything that was uneccesary due to innate tools now working, made the pyrokinesis be a note until the rework of psionics, ready for review, again.

DebugOk commented:
Worst case shit horribly breaks and we revert
